src/settings-ui/Settings.UI/ViewModels/PluginMetadataViewModel.cs

```diff
@@ -0,0 +1,63 @@
+﻿// Copyright (c) Microsoft Corporation
+// The Microsoft Corporation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+
+using System.Runtime.CompilerServices;
+
+namespace Microsoft.PowerToys.Settings.UI.ViewModels
+{
+    public class PluginMetadataViewModel : INotifyPropertyChanged
+    {
+        private PluginMetadataViewModel(string value, PluginMetadataType type)
+        {
+            _value = value;
+            _type = type;
+        }
+
+        private string _value;
+
+        public string Value => _value;
+
+        private PluginMetadataType _type;
+
+        public PluginMetadataType Type => _type;
+
+        // Do not user for item separator initialization.
+        public static PluginMetadataViewModel MetadataItem(string value, PluginMetadataType type)
+        {
+            ArgumentNullException.ThrowIfNullOrWhiteSpace(value);
```

Comment:
Instead of throwing exception, consider using a more defensive approach since we don't have control on `plugin.json` for certain plugins.

Reply:
Argh. You are right. The author can be wrong.